### DISCUSSION TITLE
[Kafka Exporter] Issue during startup - too many colons in address

### DISCUSSION THREAD

**Discussion by ghost** [mia-platform/community#499] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

v12.3.2

### Description

Hi guys,
I'm trying to configure the **Kafka Exporter** microservice to enable Metrics gathering and monitoring.

I'm unable to provide the correct configuration for the Kafka Exporter, in particular for the Brokers setting.
I've tried to follow this guide [Kafka Exporter Configuration](https://docs.mia-platform.eu/docs/fast_data/monitoring/overview#kafka-exporter-configuration) but the Kafka Exporter is not starting up, giving me this error message:

`Failed to connect to broker kafka-1.server:9093,kafka-2.server:9093,kafka-3.server.eu:9093: dial tcp: address kafka-1.server:9093,kafka-2.server:9093,kafka-3.server:9093: too many colons in address`

I'm using a Public Variable for the interpolation of the Brokers and here I'm using a fake address as an example.

I've tried many different values for the Brokers variable:
- KAFKA_BROKERS = kafka-1.server:9093,kafka-2.server:9093,kafka-3.server.eu:9093
- KAFKA_BROKERS = "kafka-1.server:9093,kafka-2.server:9093,kafka-3.server.eu:9093"
- KAFKA_BROKERS = "kafka-1.server:9093","kafka-2.server:9093","kafka-3.server.eu:9093"

but I constantly get the error above.

What am I doing wrong? Is this an error related to the value of the KAFKA_BROKER variable, or it can be caused by other settings?
Thank you!

### Actual Outcome

I'm receiving an error while trying to starting up the Kafka Exporter microservices due to a bad configuration of the KAFKA_BROKER setting although the configuration is similar to the one provided in the Mia-Platform Documentation.

### Expected Outcome

Kafka Exporter is able to connect to Kafka and is able to startup correctly.

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @ivanpasquariello !

Unfortunately, I think there is an error in how the application sets up the kafka exporter's command line arguments.

In [the kafka exporter repo](https://github.com/danielqsj/kafka_exporter), is stated that the flag `--kafka.server` supports only one broker. So, if you need to define more brokers, you have to define the flag `--kafka.server` multiple times, one for each node.

@albertotessarotto let's keep track of this issue to adjust the doc and the application. maybe I can try to open a PR to the repo and ask to include a unique flag for `bootstrap servers`
